Date: Fri Mar  4 23:28:06 CET 2005
From: Vojtech Pavlik <vojtech@suse.cz>
Subject: Fix ALPS oops on suspend/resume

=====================================================================================

ChangeSet@1.2074, 2005-03-04 23:23:39+01:00, vojtech@suse.cz
  input: Fix ALPS reconnect/init methods to use alps_get_model()
         correctly.
  
  Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>


 alps.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)


diff -Nru a/drivers/input/mouse/alps.c b/drivers/input/mouse/alps.c
--- a/drivers/input/mouse/alps.c	2005-03-04 23:26:14 +01:00
+++ b/drivers/input/mouse/alps.c	2005-03-04 23:26:14 +01:00
@@ -342,7 +342,7 @@
 	unsigned char param[4];
 	int version;
 
-	if ((priv->i = alps_get_model(psmouse, &version)) < 0)
+	if (!(priv->i = alps_get_model(psmouse, &version)))
 		return -1;
 
 	if (priv->i->flags & ALPS_PASS && alps_passthrough_mode(psmouse, 1))
@@ -384,7 +384,7 @@
 		goto init_fail;
 	memset(priv, 0, sizeof(struct alps_data));
 
-	if ((priv->i = alps_get_model(psmouse, &version)) < 0)
+	if (!(priv->i = alps_get_model(psmouse, &version)))
 		goto init_fail;
 
 	if ((priv->i->flags & ALPS_PASS) && alps_passthrough_mode(psmouse, 1))
